src/components/buttons/search.ts

```diff
@@ -39,27 +61,26 @@ export default {
 			}
 			else {
 				msg = 'MUSIC.QUEUE.TRACK_ADDED.MULTIPLE.DEFAULT';
-				extras = [resolvedTracks.length, await getGuildLocaleString(interaction.guildId, 'MISC.YOUR_SEARCH'), ''] ;
+				extras = [resolvedTracks.length.toString(), await getGuildLocaleString(interaction.guildId, 'MISC.YOUR_SEARCH'), ''] ;
 			}
 			if (!player?.connected) {
 				player = interaction.client.music.createPlayer(interaction.guildId);
 				player.handler = new PlayerHandler(interaction.client, player);
-				player.queue.channel = interaction.channel;
+				player.queue.channel = <TextChannel | VoiceChannel> interaction.channel;
 				await player.connect(interaction.member.voice.channelId, { deafened: true });
 				// Ensure that Quaver destroys the player if the user leaves the channel while Quaver is queuing tracks
 				// Ensure that Quaver destroys the player if Quaver gets timed out by the user while Quaver is queuing tracks
 				// Ensure that Quaver destroys the player if Quaver gets kicked or banned by the user while Quaver is queuing tracks
-				const timedOut = interaction.guild?.members.me.isCommunicationDisabled();
+				const me = await interaction.guild?.members.fetchMe();
+				const timedOut = me.isCommunicationDisabled();
```

Comment:
Can we do this same optimization for the if statement check cases on the other files involving us to check for timedOut?

Reply:
sorry could you rephrase that

Reply:
oh sure